src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -64,10 +80,11 @@ class CosmasGoogleCloudService(private val bucketName: String,
             return
         }
         try {
-            project.forEach { (fileId, file) ->
+            for ((fileId, fileVersion) in project) {
                 this.storage.create(
                         BlobInfo.newBuilder(this.bucketName, fileId).build(),
-                        file.toByteArray())
+                        fileVersion.toByteArray())
+                project[fileId] = fileVersion.toBuilder().clearPatches().build()
```

Comment:
Thanks for pointing this out. I think it is okay to just clear the map when the loop finishes.

Reply:
ok